lib/src/clojure_lsp/refactor/transform.clj

```diff
@@ -397,6 +397,46 @@
           (recur (first (expand-let loc false uri db)) current-edit)
           (some-> previous-edit vector))))))
 
+(defn new-defn-zloc [fn-name private? params body db]
+  (let [root (z/of-string
```

Comment:
`create-function`, `extract-function` and `convert-fn-to-defn` (`promote-fn`) now all follow the same logic for creating a `defn` node.